### PR TITLE
(BOLT-709) Error 'apply' when report identifies failure, and provide human-readable summary

### PR DESCRIPTION
**Pull request by MikaelSmith**
Ideas for how to display apply failure output.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
One example of what output would look like with
```
plan foo::simple (TargetSpec $nodes) {
  return apply($nodes) {
    file { '/tmp/does/not/exist': ensure => directory }
    notify { 'hello': }
    alert('now')
  }
}
```
screen shot 2018-07-24 at 11 56 19 am

**Comment by MikaelSmith** [comment]
Alternative where we only emit error logs as part of an error result.

**Comment by MikaelSmith** [comment]
TODO: look at puppetlabs-apply for ideas of how we might want to reduce the report to a simpler form for printing.

File: lib/bolt/error.rb [hunk 1]
```diff
@@ -50,19 +50,32 @@ def initialize(msg)
   class RunFailure < Error
     attr_reader :result_set
 
-    def initialize(result_set, action, object)
+    def generate_msg(result_set, action, object)
```
Comment (adreyer) [inline lib/bolt/error.rb]: I think we can just override `msg` and store the and grab the action/object/result_set from details

File: lib/bolt/outputter/human.rb [hunk 2]
```diff
@@ -165,6 +165,8 @@ def print_plan_info(plan)
       def print_plan_result(plan_result)
         if plan_result.value.nil?
           @stream.puts("Plan completed successfully with no result")
+        elsif plan_result.value.is_a? Bolt::ApplyFailure
```
Comment (adreyer) [inline lib/bolt/outputter/human.rb]: Lets create a ticket to revisit human output for plans that error. I suspect there is something standard we can do here.

Comment (MikaelSmith) [reply lib/bolt/outputter/human.rb]: Added BOLT-716.

File: lib/bolt/error.rb (hunk 1 above)
Comment (MikaelSmith) [reply lib/bolt/error.rb]: Overrode `to_s`, as that seems to be what everything else - inspect, message, msg - uses.

**Comment by nicklewis** [comment]
Tried this out in one of my own use cases and the failure output is :+1:. The compilation failure is better too.

**Pull request merged**